fuzz/server.c

```diff
@@ -619,7 +620,24 @@ int FuzzerTestOneInput(const uint8_t *buf, size_t len)
     out = BIO_new(BIO_s_mem());
     SSL_set_bio(server, in, out);
     SSL_set_accept_state(server);
+
+    opt = (uint8_t)buf[len-1];
+    len--;
+
     OPENSSL_assert((size_t)BIO_write(in, buf, len) == len);
+
+    if ((opt & 0x01) != 0)
+    {
+        do {
+            char early_buf[16384];
+            size_t early_len;
+            ret = SSL_read_early_data(server, early_buf, sizeof(early_buf), &early_len);
+
+            if (ret != SSL_READ_EARLY_DATA_SUCCESS)
```

Comment:
Probably this needs to be `ret == SSL_READ_EARLY_DATA_FINISH`. If you get SSL_READ_EARLY_DATA_ERROR then you should really call SSL_get_error() to decide what to do with it. It might just be an NBIO event.